ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HivePointLookupOptimizerRule.java

```diff
@@ -734,6 +721,30 @@ private RexNode useStructIfNeeded(List<? extends RexNode> columns) {
               }
               operands.remove(i);
               --i;
+            } else if (operand.getKind() == SqlKind.EQUALS) {
+              Constraint c = Constraint.of(operand);
+              if (c == null || !HiveCalciteUtil.isDeterministic(c.exprNode)) {
+                continue;
+              }
+              String ref = c.exprNode.toString();
+              stringToExpr.put(ref, c.exprNode);
+              if (inLHSExprToRHSExprs.containsKey(ref)) {
+                String expr = c.constNode.toString();
+                stringToExpr.put(expr, c.constNode);
```

Comment:
note: we have RexNode-s are comparable; this string2expr stuff is not neccessarily needed anymore

Reply:
Makes sense, I will simplify this code.